py/runtime.h

```diff
@@ -109,6 +110,46 @@ bool mp_sched_schedule(mp_obj_t function, mp_obj_t arg);
 bool mp_sched_schedule_node(mp_sched_node_t *node, mp_sched_callback_t callback);
 #endif
 
+// Handles any pending MicroPython events without waiting for an interrupt or
+// event.
+STATIC MP_ALWAYSINLINE inline void mp_event_handle_nowait(void) {
+    #if defined(MICROPY_EVENT_POLL_HOOK_FAST) && !MICROPY_PREVIEW_VERSION_2
+    // For ports still using the old macros
+    MICROPY_EVENT_POLL_HOOK_FAST
+    #else
+    mp_handle_pending(true);
+    #endif
+}
+
+// Handles any pending MicroPython events and then suspends execution until the
+// next interrupt or event.
+//
+// Note: on "tickless" ports this can suspend execution for a long time,
+// don't call unless you know an interrupt is coming to continue execution.
```

Comment:
Maybe add "and may also return early".

Reply:
Reworded both these comments to explicitly explain that it's the tick macro that can cause frequent wakeups.